### PR TITLE
improvement(network nemesis): wait nodes UN before end of nemesis

### PR DESCRIPTION
**Pull request by fgelcer**
saw a case where BlockNetwork finished, and immediately after
it ClusterHealthValidator run (and failed) because it takes
few seconds to gossip to mark the node as UN again, making the
test to fail.
Now, this nemesis will wait for all nodes to be UN before
finishing the nemesis, and allowing ClusterHealthValidator
to run on a clean cluster.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
running a job to check if this works fine:
https://jenkins.scylladb.com/view/scylla-4.3/job/scylla-4.3/job/Reproducers/job/network_nemesis_wait_nodetool_status_un/

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2123,6 +2123,17 @@ def disrupt_network_block(self):
             time.sleep(wait_time)
         finally:
             self.target_node.traffic_control(None)
+            self._wait_all_nodes_un()
```
Comment (amoskong) [inline sdcm/nemesis.py:2126]: Can we reuse self.cluster.check_nodes_up_and_normal()?

Comment (fgelcer) [reply sdcm/nemesis.py:2126]: yes, we can and PR was already updated.

**Comment by fgelcer** [comment]
on [this job](https://jenkins.scylladb.com/view/scylla-4.3/job/scylla-4.3/job/Reproducers/job/network_nemesis_wait_nodetool_status_un/2) the issue of ClusterHealthCheck failing wasn't reproduced.
changing this PR from draft to PR, and it is ready for review.

Thank you @amoskong for your suggestion, it works like a charm :-)

**Review by roydahan — APPROVED** [review]

**Pull request merged**